packages/mobile/views/dashboard/drawers/send/views/AmountView.svelte

```diff
@@ -1 +1,144 @@
+        allowedDecimals = IOTA_UNIT_MAP?.[unit?.substring(0, 1)] ?? 0
+    }
+
+    $: bigAmount = convertToRawAmount(amount, unit, asset?.metadata)
+    $: amount, validate()
+
+    onMount(() => {
+        if (submitSendOnMount) {
+            loading = true
+            onSend()
+        }
+        if ($newTransactionDetails?.type === NewTransactionType.TokenTransfer) {
+            const storedRawAmount = $newTransactionDetails?.rawAmount
+            asset = $newTransactionDetails.asset
+            tokenMetadata = asset?.metadata
+            unit = $newTransactionDetails.unit ?? tokenMetadata?.unit
+            amount = storedRawAmount
+                ? formatTokenAmountDefault(Number(storedRawAmount), asset?.metadata, unit, false)
+                : ''
+        }
+        amountInputElement.focus()
+    })
+
+    function validate(allowZeroOrNull = false): void {
+        const amountAsFloat = parseCurrency(amount)
+        const isAmountZeroOrNull = !Number(amountAsFloat)
+        // Zero value transactions can still contain metadata/tags
+        if (allowZeroOrNull && isAmountZeroOrNull) {
+            error = null
+        } else if (isAmountZeroOrNull) {
+            error = localize('error.send.amountInvalidFormat')
+        } else if (
+            (unit === asset?.metadata?.subunit ||
+                (unit === asset?.metadata?.unit && asset?.metadata?.decimals === 0)) &&
+            Number.parseInt(amount, 10).toString() !== amount
+        ) {
+            error = localize('error.send.amountNoFloat')
+        } else if (bigAmount.gt(Big(asset?.balance?.available))) {
+            error = localize('error.send.amountTooHigh')
+        } else if (bigAmount.lte(Big(0))) {
+            error = localize('error.send.amountZero')
+        } else if (!bigAmount.mod(1).eq(Big(0))) {
+            error = localize('error.send.amountSmallerThanSubunit')
+        } else {
+            error = null
+        }
+
+        if (!error) {
+            rawAmount = bigAmount.toString()
+        }
+    }
+
+    function toggleUnit(newUnit: string): void {
+        unit = newUnit
+        amountInputElement.focus()
+    }
+
+    async function onContinueClick(): Promise<void> {
+        loading = true
+        updateNewTransactionDetails({
+            type: $newTransactionDetails.type,
+            rawAmount,
+            unit,
+        })
+        const isUnlocked = await isStrongholdUnlocked()
+        if (isUnlocked) {
+            try {
+                await onSend()
+            } catch (e) {
+                loading = false
+            }
+        } else {
+            $sendRouter.next()
+        }
+    }
+</script>
+
+<div class="w-full overflow-y-auto flex flex-col flex-auto h-1 justify-between">
+    <div class="flex flex-row flex-1 items-center justify-center relative">
+        <div class="flex flex-row items-center space-x-2 px-28" on:click={() => amountInputElement.focus()}>
+            <AmountInput
```

Comment:
Maybe not so important right now but I tested the "insane rich" case again 😅  Could be a problem if you switch to the subunit?
![image](https://user-images.githubusercontent.com/6748220/203287047-6a722822-5852-4f68-a664-da545e99a1e5.png)

Reply:
indeed it needs improvement, I would leave that for later though, the entire amount needs polish because its an input field followed by a text, centering that and keeping the input large enough for mega rich is tough, we can revisit later in a polish round